src/main/java/io/confluent/idesidecar/restapi/featureflags/HttpFlagEvaluationProvider.java

```diff
@@ -0,0 +1,109 @@
+package io.confluent.idesidecar.restapi.featureflags;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.launchdarkly.sdk.LDContext;
+import com.launchdarkly.sdk.json.JsonSerialization;
+import io.confluent.idesidecar.restapi.util.WebClientFactory;
+import io.quarkus.logging.Log;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.smallrye.common.constraint.NotNull;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collection;
+import java.util.function.Consumer;
+
+/**
+ * A {@link FeatureProject.Provider} that makes a remote HTTP call to LaunchDarkly APIs to
+ * evaluate and return the feature flags for the given context.
+ */
+@RegisterForReflection
+class HttpFlagEvaluationProvider implements FeatureProject.Provider {
+
+  final String projectName;
+  final ObjectMapper objectMapper;
+  final String clientId;
+  final String fetchUri;
+  final WebClientFactory webClientFactory = new WebClientFactory();
```

Comment:
Can you `@Inject` the `WebClientFactory` object? Otherwise, we'll end up using a different instance than the rest of the application, which will be problematic if, for instance, the user uses custom SSL settings.

Reply:
Good question. The `HttpFlagEvaluationProvider` is not a bean, and I'd rather keep it not a bean. However, I take your point, and will change this so that the `WebClientFactory` instance is injected in the lowest-level bean (`FeatureFlags`) and passed down into the other objects.

Reply:
Made this change.